src/all/apt/String.cpp

```diff
@@ -329,7 +329,11 @@ StringBase& StringBase::operator=(StringBase&& _rhs_)
 {
 	if (&_rhs_ != this) {
 		if (_rhs_.isLocal()) // If rhs is local, we always can copy inside this.m_buf
+		{
+			if (m_buf == nullptr)
```

Comment:
Is this a consequence of the changes you made to the move assignment operator? I'm thinking that perhaps we should never allow `m_buf` to be `nullptr`, instead we always set it to point at the local buffer.

Reply:
Yes this is a consequence of the last change (to avoid a dealocation).
I'm was thinking to the same solution : set by default m_buf to the local storage.

For information I wrote some small unit tests for the String class. Where would be the best place for it?